### PR TITLE
Ignoring intermittent errors on EC2 api

### PR DESCRIPTION
**Pull request by guilhermef**
When starting  a node, a few times I get a 503 from the EC2 api,
this would stop the instance start.
Now It retries the api call before a hard timeout.

### CODE REVIEW

File: iaas/ec2/task.go [hunk 1]
```diff
@@ -55,14 +55,19 @@ func (t *ec2WaitTask) Run(job monsterqueue.Job) {
...
+		}
 		log.Debugf("ec2: waiting for dnsname for instance %s", machineId)
 		input := ec2.DescribeInstancesInput{
 			InstanceIds: []*string{aws.String(machineId)},
 		}
 		resp, err := ec2Inst.DescribeInstances(&input)
 		if err != nil {
-			job.Error(err)
-			break
+			log.Debugf("ec2: api error")
```
Comment (tarsisazevedo) [inline iaas/ec2/task.go]: You can use `log.Debug` here.

File: iaas/ec2/task.go [hunk 2]
```diff
@@ -55,14 +55,19 @@ func (t *ec2WaitTask) Run(job monsterqueue.Job) {
 	var notifiedSuccess bool
 	t0 := time.Now()
 	for {
+		if time.Now().Sub(t0) > time.Duration(2*timeout)*time.Second {
```
Comment (tarsisazevedo) [inline iaas/ec2/task.go]: if `timeout` is 0 there is a chance that task will not run. Is this a expected behavior?

Comment (guilhermef) [inline iaas/ec2/task.go]: There's a scenario where timeout is 0 ?

Comment (tarsisazevedo) [inline iaas/ec2/task.go]: No, sorry 😄

**Comment by tarsisazevedo** [comment]
Thanks! \o/

**Pull request merged**